plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLUtils.java

```diff
@@ -221,6 +231,23 @@ public static String makeSQLLike(String like)
         return like.replace("*", "%").replace("?", "_");
     }
 
+    public static String makeGlobFromSqlLikePattern(String sqlLikePattern) {
+
+        StringBuilder result = new StringBuilder();
+        for (int i = 0; i < sqlLikePattern.length(); i++) {
+            char charAtI = sqlLikePattern.charAt(i);
+            if (i + 1 < sqlLikePattern.length() && charAtI == '\\'
+                && (sqlLikePattern.charAt(i + 1) == '_' || sqlLikePattern.charAt(i + 1) == '%')) {
+                result.append(sqlLikePatternToGlobSymbolsMap.get(
+                    String.format("%c%c", charAtI, sqlLikePattern.charAt(i + 1))));
+                i++;
+            }
+            result.append(sqlLikePatternToGlobSymbolsMap.computeIfAbsent(String.valueOf(charAtI), Function.identity()));
```

Comment:
... and here we are only checking for `_`, `%`, and `?`, really. Since you're calling `computeIfAbsent` to avoid introducing a local variable, you will poison the map with all possible characters. You either want to use `Map#getOrDefault`, or check explicitly for these two cases with an `if` statement.

Reply:
You don't need a lookup map for such a simple function. Please refactor it.

Reply:
Totally agree about computeIfAbsent. There is no need to push new elements into map, thank you about that. But getting rid of map leads to overcomplication of code with multiply if-statement. I don't think that it is something that we want

Reply:
I agree that we don't need map for such a simple algorithm. static switch/case is much better